aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -185,50 +183,56 @@ private Completable performSynchronized(Completable completable) {
                     "Retry your request."));
         }
         LOG.info("Orchestrator lock acquired.");
-        return completable.doFinally(() -> {
-            startStopSemaphore.release();
-            LOG.info("Orchestrator lock released.");
-        });
+        return Completable.fromAction(action)
+            .doFinally(() -> {
+                startStopSemaphore.release();
+                LOG.info("Orchestrator lock released.");
+            }
+        );
     }
 
-    private DataStoreException unknownStateError(State state) {
-        return new DataStoreException(
+    private void unknownState(State state) throws DataStoreException {
+        throw new DataStoreException(
                 "Orchestrator state machine made reference to unknown state = " + state.name(),
                 AmplifyException.REPORT_BUG_TO_AWS_SUGGESTION
         );
     }
 
-    private Completable transitionToStopped() {
-        LOG.info("DataStore orchestrator stopping. Current mode = " + currentState.get().name());
-        disposables.clear();
+    private void transitionToStopped() throws DataStoreException {
         switch (currentState.get()) {
             case SYNC_VIA_API:
-                return stopApiSync().doFinally(this::stopObservingStorageChanges);
+                LOG.info("Orchestrator transitioning from SYNC_VIA_API to STOPPED");
```

Comment:
I had the logs outside the switch initially, but was seeing a ton of "Orchestrator transitioning from SYNC_VIA_API to SYNC_VIA_API" logs.  I moved them into the switch, so it would only log the 6 transitions that actually matter :)

I've been thinking about log levels and feel like the following makes sense:
 - `Verbose` should be used when logging something for every instance of a model
 - `Debug` should be used when logging something for every model type
 - `Info` should be used for general state changes in the orchestrator or any of the processors, that only result in ~1 instance of the log message